pkg/reader/proc/proc.go

```diff
@@ -164,29 +175,50 @@ func GetProcPid(pid string) (uint64, error) {
 	return strconv.ParseUint(pid, 10, 32)
 }
 
-// Returns real uid and effective uid on success. If we fail we do not
+// Returns all parsed UIDs on success. If we fail for one value we do not
 // return the overflow ID, we return the invalid UID 4294967295
 // (-1 as an unsigned integer).
 // The overflow ID is returned when the kernel decides and pass it back,
 // as it can be a valid indication of UID mapping error.
-func GetUids(status *Status) (uint32, uint32, error) {
-	ruid, err := strconv.ParseUint(status.Uids[0], 10, 32)
-	if err != nil {
-		return InvalidUid, InvalidUid, err
+func (status *Status) GetUids() ([]uint32, error) {
```

Comment:
@tixxdz 
would have it been possible to use an array `[4]uint32` instead of a slice and indicate which UIDS correspond to what in the doc? Same for GetGids

Reply:
I think the best here would have been to have 5 named return parameters (so that we get doc for free), I think the slice/array part makes the function less nice to use.

Reply:
AFAICT it can't return an empty, as all fields are explicitly initialized to invalid uids, so no need for a length check unless I'm missing something? plus we need to explicitly initialize this otherwise we may endup with uid 0, IIRC slices are generic so I don't think this is a real issue?

for doc https://github.com/cilium/tetragon/blob/main/pkg/reader/proc/proc.go#L17 , unless you are referring to function godoc? then there is too much work and issues to fix that I didn't think about the godoc to be honest, for me as long as the code is covered with e2e tests then I'm good ;-)

Reply:
> AFAICT it can't return an empty, as all fields are explicitly initialized to invalid uids, so no need for a length check unless I'm missing something?

sure! So why use a slice (since you know the size in advance) instead of an array or just return the values flat? I don't say it's wrong but it's just a bit weird interface, I would say if you use such a function, you have to trust it to not crash your program (if you don't check the length of the slice) while if you return an array or a flat 5 values return, it will guarantee by design that it cannot crash your program.

No, exactly I was searching for this kind of doc, because if you just read the function you don't know what is inside the slice.

Reply:
but anyway, it's just a small function that's no big deal but I just wondered why you chose a slice here.

Reply:
let's say due to background returning a reference to memory is perfectly fine! it is initialized and have its length set, so no issue here... in the other hand returning multiple vars even if ok with Go is weird interface, it means you are returning too much context in one single function and you probably have to split the logic there...

Reply:
Ok we are going in circles 😁 

> it is initialized and have its length set, so no issue here...

Let me show you what I mean, it's not that it doesn't work, something is not initialized or whatever, it's just using the benefits of the Go language: https://go.dev/play/p/ov0JGylWXup

<img width="784" alt="image" src="https://github.com/cilium/tetragon/assets/11256051/b5bdf296-bf3c-4546-838d-5738a0a6c483">

> returning multiple vars even if ok with Go is weird interface, it means you are returning too much context in one single function and you probably have to split the logic there...

I strongly disagree, hiding it in an array doesn't change anything from returning directly flat. And bundling this tuple of uids here totally make sense.